mtags/src/main/scala-3/scala/meta/internal/pc/PcDefinitionProvider.scala

```diff
@@ -0,0 +1,197 @@
+          defs.headOption match {
+            case Some(srcTree) =>
+              val pos = srcTree.namePos
+              pos.toLocation match {
+                case None => DefinitionResultImpl.empty
+                case Some(loc) =>
+                  DefinitionResultImpl(
+                    SemanticdbSymbols.symbolName(sym),
+                    List(loc).asJava
+                  )
+              }
+            case None =>
+              DefinitionResultImpl.empty
+          }
+        else {
+          val res = new ArrayList[Location]()
+          semanticSymbolsSorted(symbols)
+            .foreach { sym =>
+              res.addAll(search.definition(sym, params.uri()))
+            }
+          DefinitionResultImpl(
+            SemanticdbSymbols.symbolName(sym),
+            res
+          )
+        }
+      case Nil => DefinitionResultImpl.empty
+    }
+  }
+
+  @tailrec
+  private def enclosingSymbols(
+      path: List[Tree],
+      pos: SourcePosition,
+      indexed: IndexedContext
+  ): List[Symbol] = {
+    import indexed.ctx
+    path match {
+      // For a named arg, find the target `DefDef` and jump to the param
+      case NamedArg(name, _) :: Apply(fn, _) :: _ =>
+        val funSym = fn.symbol
+        if (
+          funSym.name == StdNames.nme.copy
+          && funSym.is(Synthetic)
+          && funSym.owner.is(CaseClass)
+        ) List(funSym.owner.info.member(name).symbol)
+        else {
+          val classTree = funSym.topLevelClass.asClass.rootTree
+          val paramSymbol =
+            for {
+              DefDef(_, paramss, _, _) <- tpd
+                .defPath(funSym, classTree)
+                .lastOption
+              param <- paramss.flatten.find(_.name == name)
+            } yield param.symbol
+          List(paramSymbol.getOrElse(fn.symbol))
+        }
+      case (_: untpd.ImportSelector) :: (imp: Import) :: _ =>
+        importedSymbols(imp, _.span.contains(pos.span))
+
+      case (imp: Import) :: _ =>
+        importedSymbols(imp, _.span.contains(pos.span))
+
+      // For constructor calls, return the `<init>` that was selected
+      case _ :: (_: New) :: (select: Select) :: _ =>
+        List(select.symbol)
+
+      // wildcard param
+      case head :: _ if (head.symbol.is(Param) && head.symbol.is(Synthetic)) =>
+        Nil
+
+      case head :: tl =>
+        if (head.symbol.is(Synthetic)) enclosingSymbols(tl, pos, indexed)
+        else if (head.symbol != NoSymbol) List(head.symbol)
+        else {
+          val recovered = recoverError(head, indexed)
+          if (recovered.isEmpty) enclosingSymbols(tl, pos, indexed)
+          else recovered
+        }
+      case Nil => Nil
+    }
+  }
+
+  private def recoverError(
+      tree: Tree,
+      indexed: IndexedContext
+  ): List[Symbol] = {
+    import indexed.ctx
+
+    def recoverDenot(d: Denotation): List[Symbol] = d match {
+      case multi: MultiPreDenotation =>
+        List(multi.last, multi.first)
```

Comment:
Why `last` and `first` ? Could you add a comment explaining it here?

Reply:
Initially, I missed that `MultiDenotation` might contain more than two denotations.
Reworked this part